docs/user/spark/sparksql.rst

```diff
@@ -0,0 +1,182 @@
+SparkSQL
+--------
+
+GeoMesa SparkSQL support builds upon the ``DataSet``/``DataFrame`` API present
+in the Spark SQL module to provide geospatial capabilities. This includes
+custom geospatial data types and functions, the ability to create a DataFrame
+from a GeoTools DataStore, and optimizations to improve SQL query performance.
+
+GeoMesa SparkSQL code is provided by the ``geomesa-spark-sql`` module:
+
+.. code-block:: xml
+
+    <dependency>
+      <groupId>org.locationtech.geomesa</groupId>
+      <artifactId>geomesa-spark-sql_2.11</artifactId>
+      // version, etc.
+    </dependency>
+
+Example
+^^^^^^^
+
+The following is a Scala example of connecting to GeoMesa Accumulo
+via SparkSQL:
+
+.. code-block:: scala
+
+    // DataStore params to a hypothetical GeoMesa Accumulo table
+    val dsParams = Map(
+      "instanceId" -> "instance",
+      "zookeepers" -> "zoo1,zoo2,zoo3",
+      "user"       -> "user",
+      "password"   -> "*****",
+      "auths"      -> "USER,ADMIN",
+      "tableName"  -> "geomesa_catalog")
+
+    // Create SparkSession
+    val sparkSession = SparkSession.builder()
+      .appName("testSpark")
+      .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .config("spark.kryo.registrator", classOf[GeoMesaSparkKryoRegistrator].getName)
+      .config("spark.sql.crossJoin.enabled", "true")
+      .master("local[*]")
+      .getOrCreate()
+
+    // Create DataFrame using the "geomesa" format
+    val dataFrame = sparkSession.read
+      .format("geomesa")
+      .options(dsParams)
+      .option("geomesa.feature", "chicago")
+      .load()
+    dataFrame.createOrReplaceTempView("chicago")
+
+    // Query against the "chicago" schema
+    val sqlQuery = "select * from chicago where st_contains(st_makeBBOX(0.0, 0.0, 90.0, 90.0), geom)"
+    val resultDataFrame = sparkSession.sql(sqlQuery)
+
+    resultDataFrame.show
+    /*
+    +-------+------+-----------+--------------------+-----------------+
+    |__fid__|arrest|case_number|                 dtg|             geom|
+    +-------+------+-----------+--------------------+-----------------+
+    |      4|  true|          4|2016-01-04 00:00:...|POINT (76.5 38.5)|
+    |      5|  true|          5|2016-01-05 00:00:...|    POINT (77 38)|
+    |      6|  true|          6|2016-01-06 00:00:...|    POINT (78 39)|
+    |      7|  true|          7|2016-01-07 00:00:...|    POINT (20 20)|
+    |      9|  true|          9|2016-01-09 00:00:...|    POINT (50 50)|
+    +-------+------+-----------+--------------------+-----------------+
+    */
+
+Usage
+^^^^^
+
+Because GeoMesa SparkSQL stacks on top of the ``geomesa-spark-core`` module,
+one or more of the ``SpatialRDDProvider`` implementations, as described in
+:doc:`/user/spark/core`, must be included on the classpath.
+
+As well, Spark must be configured to register ``GeoMesaSparkKryoRegistor``
```

Comment:
I think it is more confusing than this...  We need the GeoMesaSparkKryoRegistor if and only if we are passing around RDDs of SimpleFeatures.

In a strict dataframe use case, I *think* we can side step that.